stats/cloud/client.go

```diff
@@ -94,6 +100,10 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		}
 	}
 
+	if shouldAddIdempotencyKey(req) {
```

Comment:
I against it, as we may want the caller to supply its own Idempotency-Key. Also, I have a patch local here, which will move all request header preparation to here instead of in `Client.do`, but it will go after this PR merged.

Reply:
Fair enough, but the user would also be able to supply their own idempotency key even if we handle this in `NewRequest()`, it would just be done by deliberately overwriting whatever we generated there.

Reply:
Fixed.